auth/policies.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package auth
+
+import (
+	"context"
+
+	acl "github.com/ory/keto/proto/ory/keto/acl/v1alpha1"
+)
+
+// Authz represents a authorization service. It exposes
+// functionalities through `auth` to perform authorization.
+type Authz interface {
+	// Authorize checks authorization of the given `subject`. Basically,
+	// Authorize verifies that Is `subject` allowed to `relation` on
+	// `object`. Authorize returns the bool indicating the authorization
+	// of the given subject and the error. For example, the response as:
+	// `false, nil` means that `subject` has not `relation` on the `object`.
+	// Therefore, this response means that incoming request is denied.
+	Authorize(ctx context.Context, subject, object, relation string) (bool, error)
```

Comment:
There is no need to return both bool and error. Error is enough. Nil error indicates success.